ui/core/components/suggest_reforges_action.tsx

```diff
@@ -525,24 +509,26 @@ export class ReforgeOptimizer {
 									values: [
 										{ name: 'Select preset', value: 0 },
 										...[...statPresets.keys()].map(key => {
-											const ratingValue = statPresets.get(key)!;
-											const percentOrPointsValue = unitStat.convertRatingToPercent(ratingValue)!;
-											const percentValue = unitStat.equalsStat(Stat.StatMasteryRating) ? (percentOrPointsValue * this.player.getMasteryPerPointModifier()) : percentOrPointsValue;
+											const percentOrPointsValue = statPresets.get(key)!;
```

Comment:
It's a bit confusing that the keys for `statPresets` are proper Rating stats, but the values are percentages rather than Rating equivalents. Maybe that's okay, but there should some documentation clarifying that the expected units for the values are percent.

Reply:
Here's an idea for a more elegant way to handle this and keep the units always consistent with the key:
- Instead of defining `statSelectionPresets` as `Map<Stat, Map<string, number>>`, instead define it as `Array<UnitStatPresets>`, where `UnitStatPresets` is an interface with `unitStat: UnitStat` and `presets: Map<string, number>` as required fields.
- Then instead of checking `this.statSelectionPresets?.get(rootStat)`, we do `this.statSelectionPresets?.find((entry) => entry.unitStat.equals(unitStat))`.
- Now we don't have to prescribe some new convention for the units, and keep the global convention I've adopted across the front-end code where the assumed units are always given in the name of the stat itself. So `SpellHastePercent` presets would be assumed to be given in percent units, while `MasteryRating` presets would be assumed to be given in Rating units.
- Then you can just call `convertDefaultUnitsToPercent` when building the display strings to show in percent, and unit conversions will happen automatically.

It's slightly more code than what you have currently, but preserves the integrity of units always being consistent with the name of the stat, which is something I really want to enforce after spending so much time on the stats refactor.

Reply:
@NerdEgghead I liked that idea. I've made the changes to all the "presets", is this what you had in mind?